presto-main/src/main/java/com/facebook/presto/operator/aggregation/KeyValuePairs.java

```diff
@@ -80,11 +93,79 @@ public Slice serialize()
     {
         Block values = valueBlockBuilder.build();
         Block keys = keyBlockBuilder.build();
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), keys.getSizeInBytes() + values.getSizeInBytes());
+        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), keys.getSizeInBytes() + values.getSizeInBytes() + BooleanType.BOOLEAN.getFixedSize() * 3);
+
+        BooleanType.BOOLEAN.writeBoolean(blockBuilder, isMultiValue);
+        for (int i = 0; i < keys.getPositionCount(); i++) {
+            keyType.appendTo(keys, i, blockBuilder);
+            valueType.appendTo(values, i, blockBuilder);
+        }
+
+        return buildStructuralSlice(blockBuilder);
+    }
+
+    /**
+     * Serialize as a map: map<key, value>
+     */
+    public Slice toMapStackRepresentation()
```

Comment:
lets name this `toMapNativeEncoding` to match our developer guide (https://prestodb.io/docs/current/develop/types.html). It was a mistake to use the term "stack representation" in our code base.